.github/workflows/flex-build.yml

```diff
@@ -0,0 +1,82 @@
+---
+name: Flex Build
+
+"on":
+  workflow_call:
+    inputs:
+      mc:
+        description: Minecraft version
+        type: string
+      lex:
+        description: LexForge version
+        type: string
+      neo:
+        description: NeoForge version
+        type: string
+      java:
+        description: Java version
+        required: true
+        type: string
+      upload:
+        description: Upload the build artifacts
+        default: true
+        type: boolean
+      publish:
+        description: Run Gradle publish
+        default: false
+        type: boolean
+
+jobs:
+  build: # TODO: add build attestation and generate then combine gradle dependency graphs for SBOM
+    name: Build ${{ inputs.mc }}
+    runs-on: blacksmith-2vcpu-ubuntu-2204
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 1
+
+      - name: Cache build
+        id: cache
+        uses: useblacksmith/cache@v5
+        with:
+          path: build
+          key: build-${{ inputs.mc }}-${{ hashFiles('./*') }}
```

Comment:
This is not great and should exclude files like release-please-config.json, README.MD and LICENSE, but besides those all other files affect ci, code or build system and changes should satisfy a rebuild of everything